### PR TITLE
[MM-41785] Show error in invitation popin when email can't be send

### PR DESCRIPTION
**Pull request by JulienTant**
#### Summary
This PR allows the invitation email to return an error when having trouble sending an email.

I allowed us to have two different messages:
- If the settings are the default settings, we can have a message suggesting that this might be the issue (`SMTP is not configured in System Console` taken from the PR)
- one for the other cases (`Error while sending the email`).

Extra:
- The Server struct was bound to the `*email.Service` implementation, I decided to add an `email.ServiceInterface` to give us the ability to mock sending email in the tests.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-41785

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: app/email/service.go [hunk 1]
```diff
@@ -114,7 +116,55 @@ func (es *Service) setUpRateLimiters() error {
...
 	}
 
-	es.PerHourEmailRateLimiter = perHourRateLimiter
-	es.PerDayEmailRateLimiter = perDayRateLimiter
+	es.perHourEmailRateLimiter = perHourRateLimiter
+	es.perDayEmailRateLimiter = perDayRateLimiter
 	return nil
 }
+
+type ServiceInterface interface {
```
Comment (AGMETEOR) [inline app/email/service.go:124]: +1

**Review by AGMETEOR — APPROVED** [review]
LGTM

**Review by julmondragon — APPROVED** [review]
Looks good!

**Comment by marianunez** [comment]
@JulienTant is there a webapp change for this? Would it be possibly to distinguish if the error is because it is not configured (aka set to default settings) vs failed to sent while properly configured?

**Comment by JulienTant** [comment]
> is there a webapp change for this?

No, the error messages in the popin comes from the server

> Would it be possibly to distinguish if the error is because it is not configured (aka set to default settings) vs failed to sent while properly configured?

I'm doing this check here: https://github.com/mattermost/mattermost-server/blob/MM-41785/smtp-err/app/team.go#L1334-L1338

**Comment by marianunez** [comment]
> I'm doing this check here: [https://github.com/mattermost/mattermost-server/blob/MM-41785/smtp-err/app/team.go#L1334-L1338](https://github.com/mattermost/mattermost-server/blob/MM-41785/smtp-err/app/team.go?rgh-link-date=2022-02-24T23%3A48%3A35Z#L1334-L1338)

💯  Perfect, thanks for clarifying!

**Comment by JulienTant** [comment]
Test server ready, with SMTP unlocked: https://julien-test.test.mattermost.cloud/

**Review by stevemudie — APPROVED** [review]
Worked as expected. Looks good @JulienTant!

Screen Shot 2022-03-02 at 3 49 35 PM

**Pull request merged**